coordinator/src/test/java/org/apache/uniffle/coordinator/BasicAssignmentStrategyTest.java

```diff
@@ -147,4 +149,79 @@ public void testAssignWithDifferentNodeNum() {
     assertTrue(serverNodes.contains(sn2));
     assertTrue(serverNodes.contains(sn3));
   }
+
+  @Test
+  public void testAssignmentShuffleNodesNum() {
+    Set<String> serverTags = Sets.newHashSet("tag-1");
+
+    for (int i = 0; i < 20; ++i) {
+      clusterManager.add(new ServerNode("t1-" + i, "", 0, 0, 0,
+          20 - i, 0, serverTags, true));
+    }
+
+    /**
+     * case1: user specify the illegal shuffle node num(<0)
+     * it will use the default shuffle nodes num when having enough servers.
+     */
+    PartitionRangeAssignment pra = strategy.assign(100, 10, 1, serverTags, -1);
```

Comment:
Could you add a case when assignShuffleServer is equal to 0?

Reply:
update